### ISSUE TITLE
Feature Request: should be able to add runtimes as devDependencies

### ISSUE THREAD

**Issue by DickSmith** [NativeScript/nativescript-cli#4511]
**Is your feature request related to a problem? Please describe.**
Several, both direct and indirect.

Haven't checked them recently, as I have several work arounds in my repos as they've arisen, but some of the following have occurred or continue to occur, due to the runtimes not being an actual dev dependency:

1. In the past, the current `"version"` property cannot accept all valid `npm` syntax there. For it to be able to accept stings like `rc` and `latest` this appears to have to be manually added to the CLI project, and seems fragile to maintain. However, many other, valid strings, cannot be used here, like using a manually packed tgz and using the `"file:....." syntax. Having to add `--framework-path` every time a custom runtime needs to be used is tedious, and has even been prone to errors in the past. It is also difficult to maintain and easy to forget.
2. While not as bad as in the past, it seems redundant to have to do an `npm install` each time the `platform/*` is created fresh, this can also cause weird behaviours in npm itself, with the way it rebuilds the dependencies, and reruns some lifecycle hooks. Doing a `npm i` and then `npm un` just introduces too many edge cases, and ideally the desired version of the runtimes can be installed along with the other devDependencies, and simply copped over when necessary. The current setup circumvents `npm`s normal processes, and has risks for maintenance due to subtleties in the way `npm i` works when installing fresh, or when it is just installing/uninstalling a single package, and if a package.lock is present  from the first step.
3. Haven't tried it recently, but I believe that in the past using `next` or `rc` in those fields then gets replaced with the specific version after it has been installed, which may not be the desired behaviour depending on why those are being used,
4. I haven't tried it recently, but not sure if npm syntax like `~` `^` or other operators work in that string, if they are they may have been prone in the past to being replaced by a specific version, like `rc` and `latest` have been, which may not be the desired behaviour.
5. Cannot use commands like `npm outdated` to see if there is an available update, which is often useful for checking for updated packages.

***tl;dr:*** the current set up has been hard to maintain and work around on large, long-term projects, especially when custom runtimes need to be used. Particularities (or even flaws/bugs) of NPM also create additional problems, since it circumvents the workflows of NPM. The runtimes are a `devDependencie` in every sense, so doing a `npm i X` followed by a `npm un X` each time the platform needs to be built fresh is both unnecessary and non-standard.

**Describe the solution you'd like**
I can understand maintaining the the current setup, for easy-of-use and backwards compatibility, but the CLI should prioritize using the runtime if specified in devDependencies.

I'm aware that this circumvents a bunch of the "checking" to make sure that the runtimes and the CLI and other modules are  within compatible versions, but for now can just be an "advanced" usage. In the past I'd tried to manually make some of these changes myself, but there seems to be a lot of complexity around some of these things in the repo, and I haven't had the time to really dig in there.

**Describe alternatives you've considered**
Really, long term I see no reason the "version" tags even need to be used, rather doing a `tns add X` should just add it to the `devDependencies`, but at very least, to support more advanced use cases, having the ability to set this in the devDependeny and make that the version that is prioritized, will make it much easier to maintain advanced use cases.

**Additional context**
None currently.

**Comment by janoshrubos** [comment]
Runtimes are placed in `devDependencies` now!

```
"devDependencies": {
    "@nativescript/ios": "8.0.0",
    ...
    ...
    ...
 }
 ```
